controlplane/test/api-keys.test.ts

```diff
@@ -83,3 +91,192 @@ describe('API Keys', (ctx) => {
+      subgraphTargetIds: [],
+      federatedGraphTargetIds: [],
+      allowAllResources: true,
+    });
+    expect(response.response?.code).toBe(EnumStatusCode.OK);
+
+    await server.close();
+  });
+
+  test('Should not be able to create an api key when no resources are selected and all resources is not selected', async (testContext) => {
+    const { client, users, server } = await SetupTest({ dbname, enabledFeatures: ['rbac'] });
+
+    const response = await client.createAPIKey({
+      name: uid(8),
+      expires: ExpiresAt.NEVER,
+      userID: users.adminAliceCompanyA.userId,
+      subgraphTargetIds: [],
+      federatedGraphTargetIds: [],
+      allowAllResources: false,
+    });
+    expect(response.response?.code).toBe(EnumStatusCode.ERR);
+    expect(response.response?.details).toBe('Can not create an api key without associating it with any resources.');
+
+    await server.close();
+  });
+});
+
+describe('Create API Keys as developers with RBAC enabled', (ctx) => {
+  beforeAll(async () => {
+    dbname = await beforeAllSetup();
+  });
+
+  afterAll(async () => {
+    await afterAllSetup(dbname);
+  });
+
+  test('Should not be able to create an api key with resources the user doesnt have access', async (testContext) => {
+    const { client, users, server, authenticator } = await SetupTest({
+      dbname,
+      enabledFeatures: ['rbac'],
+      enableMultiUsers: true,
+    });
+
+    const subgraphName = genID('subgraph1');
+    const label = genUniqueLabel();
+
+    const createSubgraphResp = await client.createFederatedSubgraph({
+      name: subgraphName,
+      namespace: 'default',
+      labels: [label],
+      routingUrl: 'http://localhost:8080',
+    });
+
+    expect(createSubgraphResp.response?.code).toBe(EnumStatusCode.OK);
+
+    const getSubgraphResp = await client.getSubgraphByName({
+      name: subgraphName,
+      namespace: 'default',
+    });
+
+    expect(getSubgraphResp.response?.code).toBe(EnumStatusCode.OK);
+    expect(getSubgraphResp.graph).toBeDefined();
+    expect(getSubgraphResp.graph?.targetId).toBeDefined();
+
+    const ids: string[] = [];
+    if (getSubgraphResp.graph) {
+      ids.push(getSubgraphResp.graph.targetId);
+    }
+
+    authenticator.changeUser(TestUser.devJoeCompanyA);
+
+    const response = await client.createAPIKey({
+      name: uid(8),
+      expires: ExpiresAt.NEVER,
+      userID: users.devJoeCompanyA?.userId,
+      subgraphTargetIds: ids,
+      federatedGraphTargetIds: [],
+      allowAllResources: false,
+    });
+
+    expect(response.response?.code).toBe(EnumStatusCode.ERROR_NOT_AUTHORIZED);
+    expect(response.response?.details).toBe(
+      'You are not authorized to perform the current action as RBAC is enabled. Please communicate with the organization admin to gain access.',
+    );
+
+    await server.close();
+  });
+
+  test('Should not be able to create an api key with selected all resources', async (testContext) => {
```

Comment:
> Should not be able to create an api key as developer that has access to all resources

I think this is clearer.

Reply:
Feel that will mislead, because as a developer they can if they have access to all resources but they wont have that option.

Rephrased it to "Should not be able to create an api key as a developer with all resources option selected"

Reply:
ok